src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -6,77 +6,291 @@ description: Learn how to add a single Flutter screen to your existing iOS app.
 
 This guide describes how to add a single Flutter screen to an existing iOS app.
 
-## Write code to use FlutterViewController from your host app
+## Start a FlutterEngine and FlutterViewController
 
-The proper place to do this is specific to your host app. Here is an example
-that makes sense for the blank screen of the host app generated by Xcode.
+Launching a Flutter screen from an existing iOS comprises of starting a
+[`FlutterEngine`]({{site.api}}/objcdoc/Classes/FlutterEngine.html)
+and a [`FlutterViewController`]({{site.api}}/objcdoc/Classes/FlutterViewController.html).
 
-First declare your app delegate to be a subclass of `FlutterAppDelegate`. Then define a FlutterEngine property, which helps you to register a plugin without a FlutterViewController instance.
+{{site.alert.secondary}}
+  The `FlutterEngine` serves as a host to the Dart VM and your Flutter runtime
+  and the `FlutterViewController` attaches to a `FlutterEngine` to pass UIKit
+  input events into Flutter and to display frames rendered by the `FlutterEngine`.
+{{site.alert.end}}
 
-In `AppDelegate.h`:
+The `FlutterEngine` may have the same lifespan as your `FlutterViewController`
+or outlive your `FlutterViewController`.
 
-```objective-c
+{{site.alert.tip}}
+It's generally recommended pre-warm a long-lived `FlutterEngine` for your
+application. This way,
+
+- Flutter screens will render their first frames faster.
+- Your Flutter and Dart state could outlive one `FlutterViewController`.
```

Comment:
I'm always a bit thrown by what `Flutter` means in this context.  Do we just mean `Flutter-specific application state`?

Reply:
"could" or "will"?

Reply:
@dnfield I'm actually just kinda thinking anything you can conceptually assign to the term 'Flutter' and that has a state. Can you think of counter examples? 

Replaced could with will

Reply:
Is `Flutter` here the Flutter framework?  The Flutter engine excluding the embedding?  The embedding?  Plugin code (dart/platform)?

Reply:
Doesn't the sentence apply the same way for all those definitions? 

Tangent: I don't think the word embedding means anything for people outside the team and I'm tempted to keep it so for as long as we can.

Reply:
I guess a way of looking at this is: here you use `Flutter` in a way that excludes`FlutterViewController`.  Later in the document you're using `Flutter` in a way that explicitly means showing the `FlutterViewController`.

Reply:
Isn't it a bit like having the sentences:

- Having a backup of your movie will make it outlive one VHS tape
- 'showMovie'

and then asking whether readers will be struggling understanding whether 'movie' here encompasses the mylar tape ribbons or not :)

It's likely immaterial whether they assume one way or another.